powertools-cloudformation/src/test/java/software/amazon/lambda/powertools/cloudformation/handlers/RuntimeExceptionThrownHandler.java

```diff
@@ -0,0 +1,24 @@
+package software.amazon.lambda.powertools.cloudformation.handlers;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.events.CloudFormationCustomResourceEvent;
+import software.amazon.lambda.powertools.cloudformation.AbstractCustomResourceHandler;
+import software.amazon.lambda.powertools.cloudformation.Response;
+
+public class RuntimeExceptionThrownHandler extends AbstractCustomResourceHandler {
+
+    @Override
+    protected Response create(CloudFormationCustomResourceEvent event, Context context) {
+        throw new RuntimeException("failure");
+    }
+
+    @Override
+    protected Response update(CloudFormationCustomResourceEvent event, Context context) {
+        throw new RuntimeException("failure");
+    }
+
+    @Override
+    protected Response delete(CloudFormationCustomResourceEvent event, Context context) {
+        return null;
```

Comment:
what happens if there is an exception in the delete? do we have an IT for this case?

Reply:
It _should_ be the same as for an UPDATE (echo back request phyId) but I don't think we have coverage. Is there a standard way of doing parameterized tests in Java these days - I think some of these tests would work quite nicely parameterized out and let us more easily understand the intent - here for instance you want to assert the behaviour is the same for UPDATE/DELETE for both exceptional and null responses.

Reply:
https://junit.org/junit5/docs/current/api/org.junit.jupiter.params/org/junit/jupiter/params/ParameterizedTest.html

Reply:
> Please can we add the following tests:
> 
>     * `CREATE` with a user provided `physicalResourceId` via `Response.success(physicalResourceId)` and `Response.failed(physicalResourceId)`- validate that the `physicalResourceId` is sent to CF
> 
>     * `UPDATE` with a user provided `physicalResourceId` via `Response.success(physicalResourceId)` and `Response.failed(physicalResourceId)`- validate that the `physicalResourceId` is sent to CF
> 
>     * `DELETE` with a user provided `physicalResourceId` via `Response.success(physicalResourceId)` and `Response.failed(physicalResourceId)`- validate that the `physicalResourceId` is sent to CF
> 
> 
> Adding these tests will ensure that we are indeed sending the provided `physicalResourceId` to Cloudformation, and they will help us catch any future regressions

Suggest doing this with a parameterized test per @jeromevdl 's link above; it'll make it easier to understand the intent - "C/U/D all pass the physicalResourceId along to CF on success"